src/component/project/Project/hooks/useProjectForm.ts

```diff
@@ -0,0 +1,85 @@
+import { useEffect, useState } from 'react';
+import useProjects from '../../../../hooks/api/getters/useProjects/useProjects';
+import { IPermission } from '../../../../interfaces/project';
+
+export interface ICheckedPermission {
+    [key: string]: IPermission;
+}
+
+const useProjectForm = (
+    initialProjectId = '',
+    initialProjectName = '',
+    initialProjectDesc = ''
+) => {
+    const { projects } = useProjects();
+    const [projectId, setProjectId] = useState(initialProjectId);
+    const [projectName, setProjectName] = useState(initialProjectName);
+    const [projectDesc, setProjectDesc] = useState(initialProjectDesc);
+    const [errors, setErrors] = useState({});
+
+    useEffect(() => {
+        setProjectId(initialProjectId);
+    }, [initialProjectId]);
+
+    useEffect(() => {
+        setProjectName(initialProjectName);
+    }, [initialProjectName]);
+
+    useEffect(() => {
+        setProjectDesc(initialProjectDesc);
+    }, [initialProjectDesc]);
+
+    const getProjectPayload = () => {
+        return {
+            id: projectId,
+            name: projectName,
+            description: projectDesc,
+        };
+    };
+
+    const validateIdUniqueness = () => {
```

Comment:
This is not good enough to validate the id. We need to call an endpoint to do this on the server side. The existing project performs the validation like this: 

```
    const validateId = async (id: string) => {
        if (editMode) return true;

        const e = { ...errors };
        try {
            await props.validateId(id);
            e.id = undefined;
        } catch (err: any) {
            e.id = err.message;
        }

        setErrors(e);
        if (e.id) return false;
        return true;
    };

```